tactics/tacticals.ml

```diff
@@ -734,13 +732,11 @@ module New = struct
   let case_nodep_then_using =
     general_elim_then_using gl_make_case_nodep false
 
-  let pf_constr_of_global ref tac =
-    Proofview.Goal.enter { enter = begin fun gl ->
-      let env = Proofview.Goal.env gl in
-      let sigma = Tacmach.New.project gl in
-      let (sigma, c) = Evd.fresh_global env sigma ref in
-      let c = EConstr.of_constr c in
-      Proofview.Unsafe.tclEVARS sigma <*> (tac c)
-    end }
+  let pf_constr_of_global ref =
```

Comment:
Do I understand correctly that since `ref` is a global, typing it in the global environment is the same thing is typing it in the current goal context?

Still, isn't this changing the semantics of `pf_constr_of_global` since it calls `Evd.fresh_goal` only once for all subgoals? Sorry, I don't know what `Evd.fresh_global` does.